plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -206,13 +207,17 @@ export function createFetchTemplateAction(options: {
         } else {
           renderFilename = renderContents = !nonTemplatedEntries.has(location);
         }
+
         if (renderFilename) {
           localOutputPath = renderTemplate(localOutputPath, context);
         }
+
+        if (containsSkippedContent(localOutputPath)) {
+          continue;
+        }
+
         const outputPath = resolveSafeChildPath(outputDir, localOutputPath);
-        // variables have been expanded to make an empty file name
-        // this is due to a conditional like if values.my_condition then file-name.txt else empty string so skip
-        if (outputDir === outputPath) {
+        if (fs.existsSync(outputPath)) {
```

Comment:
I have added this condition to avoid issues when the outputPath exists (that means that the file has been skipped and the output path is a directory, not a file)